### DISCUSSION TITLE
v3.1.197.0

### DISCUSSION THREAD

**Discussion by doug24** [dnGrep/dnGrep#787] [General]
[![v3.1.197.0](https://img.shields.io/github/downloads/dnGrep/dnGrep/v3.1.197.0/total.svg)](https://github.com/dnGrep/dnGrep/releases/tag/v3.1.197.0)

### Scripts

**Scripting** is new in this release.  Scripts can be used to automate all of the properties and functions on the dnGrep main window.  Create and save scripts, run them from the Scripts menu, or call them from a command line argument. Scripts can search, replace, sort, create reports, copy results, move, copy, or delete files, use bookmarks, and call external commands. See the [scripting commands](https://github.com/dnGrep/dnGrep/wiki/Scripting%20Commands) for more information.  

### Issues

**#757 Move to recycle bin** Added new context menu item to the search results to move files to the recycle bin.

**#759 Pin bookmark window** Added a "pin" button to the bookmark window so it will stay open after selecting a bookmark to use.

**#765 PDF page number**  Added an option to show the PDF page number in place of the line number in the results tree and preview window. The page number is repeated for each matching line on the page. These numbers match the page number controls in PDF readers, but not the number on page on the document page if there are non-numbered pages at the start the document.

**#768 Bookmark name** Added an optional bookmark name field to bookmarks.  The bookmark name may be used to organize and find bookmarks, and are required to use a bookmark in a script.  The order of the bookmarks in the bookmark window is now saved, and you can reorder them to organize them as you like.

**#769 Everything quotes** Fixed bugs adding quotes to paths in Everything search terms.

**#782 Explorer context menu** Added a new context menu item to the search results to open the Explorer context menu for the selected files.

**Delete files option**  Added a new option to specify what the Delete files command does - send the files to the recycle bin, or permanently delete the files.  The new default is to send to the recycle bin.  This option only affects the Delete files command from the "More..." menu or script, not the new "Move files to recycle bin" context menu item.

### Translations

**Added català, eesti and italiano translations.** Thank-you for the new translations.  

These and many of the other translations need help to complete them.  You are welcome and encouraged to complete or improve any of the existing translations, see [Translate dnGrep](https://github.com/dnGrep/dnGrep/wiki/Translate-dnGrep) for more information. Do as much or as little as you like: there are some translations missing just a few phrases, and some new languages that need to be more complete before being included in a dnGrep release. 

### Checksum information

Name: dnGREP.3.1.197.x64.msi
SHA256: 47d9faf3d241bd8e8885b5e671530fc3f80a38ed335f0c5ee870a990f74b5c2a

Name: dnGREP.3.1.197.x86.msi
SHA256: 8f7bcf5eb347bd45670108f08fc955b5071d2589f5cc48a7fd3b981e4505feff

Free code signing provided by [SignPath.io](https://signpath.io/), certificate by [SignPath Foundation](https://signpath.org/).

<hr /><em>This discussion was created from the release <a href='https://github.com/dnGrep/dnGrep/releases/tag/v3.1.197.0'>v3.1.197.0</a>.</em>

**Comment by piranah** [comment]
Guys.....one word .....fffffffffffff**********************cccccccccccccccccckkkkkkkkkkkkkkkkkkk......exactly what I was waiting for...AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME ! AWESOME !
